contracts/rewards/AutoCompoundingRewards.sol

```diff
@@ -281,10 +298,51 @@ contract AutoCompoundingRewards is IAutoCompoundingRewards, ReentrancyGuardUpgra
         emit ProgramEnabled({ pool: pool, status: status, remainingRewards: p.remainingRewards });
     }
 
+    /**
+     * @inheritdoc IAutoCompoundingRewards
+     */
+    function setAutoTriggerCount(uint256 newAutoTriggerCount) external onlyAdmin {
+        uint256 prevAutoTriggerCount = _autoTriggerCount;
+        if (prevAutoTriggerCount == newAutoTriggerCount) {
+            return;
+        }
+
+        _autoTriggerCount = newAutoTriggerCount;
+
+        emit AutoTriggerCountUpdated({
+            prevAutoTriggerCount: prevAutoTriggerCount,
+            newAutoTriggerCount: newAutoTriggerCount
+        });
+    }
+
+    /**
+     * @inheritdoc IAutoCompoundingRewards
+     */
+    function trigger() external nonReentrant {
+        address[] memory values = _pools.values();
```

Comment:
Keep in mind that this loads all the `_pools` to the memory and if there are too many (e.g., tens of thousands) - this will exceed the block gas limit. This isn't an issue, but we should still consider not loading all the pools to reduce the gas cost of this function, especially when `values` are only accessed by index anyway...

Reply:
Well, the alternative - calling `_pools.at` - yields an **internal** function call, because `EnumerableSetUpgradeable.AddressSet` is used as an internal library in this contract.

On the other hand, it essentially passes `_pools` to function `at`, which might yield an additional storage-read per iteration (I'm only 50% sure of that TBH).

So possibly, changing to this scheme might also increase the gas cost.
In fact, there's a good chance that the internal function call per iteration alone (i.e., even without the additional storage-read per iteration) will increase the gas cost in the same manner as reading all the values into memory (at least in the specific case when `_autoTriggerCount` is as large as the number of pools).

We probably need to do some measurements, but I overall tend to accept this suggestion so fixed; please re-review.